libr/core/visual.c

```diff
@@ -4262,26 +4262,38 @@ R_API int r_core_visual(RCore *core, const char *input) {
 			} else {
 				ch = r_cons_readchar ();
 			}
-#ifndef __WINDOWS__
-			if (IS_PRINTABLE (ch) || ch == '\t' || ch == '\n') {
-				tcflush (STDIN_FILENO, TCIFLUSH);
-			} else if (ch == 0x1b) {
-				char chrs[2];
-				int chrs_read = 1;
-				chrs[0] = r_cons_readchar ();
-				if (chrs[0] == '[') {
-					chrs[1] = r_cons_readchar ();
-					chrs_read++;
-					if (chrs[1] >= 'A' && chrs[1] <= 'D') { // arrow keys
-						tcflush (STDIN_FILENO, TCIFLUSH);
-						// Following seems to fix an issue where scrolling slows
-						// down to a crawl after some time mashing the up and down
-						// arrow keys
-						r_cons_set_raw (false);
-						r_cons_set_raw (true);
+#ifdef __WINDOWS__
+			if (I->vtmode == 2 && !I->term_xterm) {
```

Comment:
What is vtmode == 2? What does it represent?

Also here I would try to extract this in a separate function, so the flow is easier to follow without all the ifdefs.
Something like:
```C
#ifdef __WINDOWS__
static bool is_vt_compatible(void) {
   return I->vtmode == 2 && !I->term_xterm
}
#else
static bool is_vt_compatible(void) {
   return true;
}
```

and then here you can just do `if (is_vt_compatible ()) {` and it works both on linux and on windows. The name `is_vt_compatible` is probably not appropriate, as I'm not sure what `I->vtmode == 2` does.

Reply:
> What is vtmode == 2? What does it represent?

I'm not sure either (ask @GustavoLCR) but what I know is that it enables mouse support in Visual panels (`V!`).

Reply:
> > What is vtmode == 2? What does it represent?
> 
> I'm not sure either (ask @GustavoLCR) but what I know is that it enables mouse support in Visual panels (`V!`).

vtmode == 2 means the terminal sends vt escape sequences as input

Reply:
This raises the question of what happens when vtmode != 2, but https://docs.microsoft.com/en-us/windows/console/setconsolemode and https://docs.microsoft.com/en-us/windows/console/reading-input-buffer-events seems to explain sufficiently.

Reply:
> Also here I would try to extract this in a separate function, so the flow is easier to follow without all the ifdefs.

Ok 19f1e733ccbf5b652c7e5dc1737969c6507ca472